src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQueryReadDataset.java

```diff
@@ -193,63 +197,136 @@ private SQLReadResult readInternal(SQLReadRequest readRequest,
       return SQLReadResult.unsupported(datasetName);
     }
 
-    // Get source table instance
-    Table srcTable = bigQuery.getTable(sourceTableId);
-
-    // Get source table instance
-    Table destTable = bigQuery.getTable(sourceTableId);
-
-    //Get Source Table Object : will be used for metadata like TABLE.TYPE and Time Partitioning
     Table sourceTable;
     try {
       sourceTable = bigQuery.getTable(sourceTableId);
     } catch (BigQueryException e) {
       throw new IllegalArgumentException("Unable to get details about the BigQuery table: " + e.getMessage(), e);
     }
 
-    // Get query job configuration based on wether the job is an insert or update/upsert
-    QueryJobConfiguration.Builder queryConfigBuilder = getQueryBuilder(sourceTable, sourceTableId,
-                                                                       destinationTableId,
-                                                                       fields,
-                                                                       sourceConfig.getFilter(),
-                                                                       sourceConfig.getPartitionFrom(),
-                                                                       sourceConfig.getPartitionTo());
+    // no Filter + no view + no material + no external
+    StandardTableDefinition tableDefinition = Objects.requireNonNull(sourceTable).getDefinition();
+    Type type = tableDefinition.getType();
+    if (!(type == Type.VIEW || type == Type.MATERIALIZED_VIEW || type == Type.EXTERNAL)
+        && sourceConfig.getFilter() == null) {
+      // TRY SNAPSHOT
+      JobConfiguration jobConfiguration = getBQSnapshotJobConf(sourceTableId, destinationTableId);
+      SQLReadResult snapshotResult = executeBigQueryJob(jobConfiguration, sourceTable, sourceTableId,
+                                                        BigQueryJobType.COPY_SNAPSHOT);
+      if (snapshotResult.isSuccessful()) {
+        return snapshotResult;
+      }
+      LOG.warn("Big Query Snapshot process (copy job) failed. Going to fallback to basic Query Job which might" +
```

Comment:
nit: Change to `Big Query Snapshot process used for direct bigquery read failed. Using fallback table copy strategy.`